### PR TITLE
feat: [Cadence Schedules] Add proto mappers for core schedule types

### PR DESCRIPTION
**Pull request by abhishekj720**
**What changed?**
Add bidirectional proto mappers (From*/To*) for core schedule types:
- ScheduleSpec, StartWorkflowAction, ScheduleAction, and SchedulePolicies, along with enum mappers for ScheduleOverlapPolicy and ScheduleCatchUpPolicy.

Key changes:
- Add local time/duration conversion helpers for mapping value-type `time.Time` and `time.Duration` fields to/from
  protobuf `Timestamp` and `Duration` messages
- Add enum mappers for `ScheduleOverlapPolicy` and `ScheduleCatchUpPolicy`
- Add From/To mappers for `ScheduleSpec`, `StartWorkflowAction`,`ScheduleAction`, and `SchedulePolicies`
- Add round-trip tests (nil, empty, populated) for all mappers

**Why?**
The mapper layer bridges protobuf wire types (`cadence-idl/go/proto/api/v1`) and internal service types (`common/types`). Without it, the frontend gRPC handler cannot deserialize schedule RPCs into types the service layer understands. 

Custom time/duration helpers were needed because schedule types use value-type `time.Time` and `time.Duration` fields whereas existing codebase helpers operate on `*int64` unix-nano pointers.

**How did you test it?**
go test ./common/types/mapper/proto/ -run ...
All 4 round-trip tests pass (nil → nil, empty → empty, populated → populated).

**Potential risks**
Low. This is a pure additive change, no existing code paths are modified.

**Release notes**
N/A (intermediate building block, not a user-facing feature yet)

**Documentation Changes**
N/A
---

## Reviewer Validation

**PR Description Quality** (check these before reviewing code):

- [ ] **"What changed"** provides a clear 1-2 line summary
  - [ ] Project Issue is linked
- [ ] **"Why"** explains the full motivation with sufficient context
- [ ] **Testing is documented:**
  - [ ] Unit test commands are included (with exact `go test` invocation)
  - [ ] Integration test setup/commands included (if integration tests were run)
  - [ ] Canary testing details included (if canary was mentioned)
- [ ] **Potential risks** section is thoughtfully filled out (or legitimately N/A)
- [ ] **Release notes** included if this completes a user-facing feature
- [ ] **Documentation** needs are addressed (or noted if uncertain)

### CODE REVIEW

File: common/types/mapper/proto/schedule_test.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+	"github.com/uber/cadence/common/types"
+	"github.com/uber/cadence/common/types/testdata"
+)
+
+func TestScheduleSpec(t *testing.T) {
+	for _, item := range []*types.ScheduleSpec{nil, {}, &testdata.ScheduleSpec} {
+		assert.Equal(t, item, ToScheduleSpec(FromScheduleSpec(item)))
+	}
+}
```
Comment (c-warren) [inline common/types/mapper/proto/schedule_test.go:39]: These tests are reliant on the test data being comprehensive enough to test the various permutations that could occur. We've seen in the past incidents caused by a lack of data here. 

I'd recommend following the pattern we introduced to run many tests with randomly injected data using the go fuzz package as we'll have good coverage of weird issues like:
* null/nil items getting handled incorrectly in round trips
* unintended differences between from/to across the full suite of potential values for a given data type

Here is an example PR that introduced it in the go-client: https://github.com/cadence-workflow/cadence-go-client/pull/1457

I'm planning on introducing this to this repo for all of our proto mappers as well - but if you're adding new ones we might as well start here!

File: common/types/mapper/proto/helpers.go [hunk 2]
```diff
@@ -97,6 +97,39 @@ func timestampToTime(t *gogo.Timestamp) *time.Time {
 	return &time
 }
 
+func timeToTimestampVal(t time.Time) *gogo.Timestamp {
+	if t.IsZero() {
+		return nil
+	}
+	nanos := t.UnixNano()
+	return unixNanoToTime(&nanos)
+}
```
Comment (c-warren) [inline common/types/mapper/proto/helpers.go]: We already have timeToTimestamp - any reason you need a new function for this?

Comment (abhishekj720) [reply common/types/mapper/proto/helpers.go]: timeToTimestamp takes *time.Time (pointer) and nil-checks it. Our schedule types use time.Time (value)

File: common/types/mapper/proto/schedule_test.go (hunk 1 above)
Comment (abhishekj720) [reply common/types/mapper/proto/schedule_test.go:39]: Good suggestion! I'll add fuzz tests for the schedule mappers. The schedule types do need custom fuzz functions for enums (to generate valid values) and time.

Comment (abhishekj720) [reply common/types/mapper/proto/schedule_test.go:39]: https://github.com/cadence-workflow/cadence/pull/7770/changes

File: common/types/mapper/proto/helpers.go (hunk 2 above)
Comment (c-warren) [reply common/types/mapper/proto/helpers.go]: Even if we have a value we could pass the pointer to timeToTimestamp. I would err on the side of consistency/simplicity in our mappers and have as few ways to convert between types as possible. If we need the `IsZero` within timeToTimestamp then I'd move that in there.

Comment (abhishekj720) [reply common/types/mapper/proto/helpers.go]: makes sense. Folded IsZero into timeToTimestamp and removed timeToTimestampVal. 

Kept timestampToTimeVal because timestampToTime returns *time.Time while schedule fields use time.Time values: changing its return type would break all existing callers, and the value variant also needs .UTC() for round-trip correctness.
